src/Sylius/Component/Core/Shipping/Checker/Eligibility/ZoneEligibilityChecker.php

```diff
@@ -30,14 +29,12 @@ public function __construct(private ZoneMatcherInterface $zoneMatcher)
 
     public function isEligible(ShippingSubjectInterface $shippingSubject, ShippingMethodInterface $shippingMethod): bool
     {
-        Assert::isInstanceOf($shippingSubject, ShipmentInterface::class);
-        Assert::isInstanceOf($shippingMethod, CoreShippingMethodInterface::class);
-
-        $shippingAddress = $shippingSubject->getOrder()?->getShippingAddress();
-        if (null === $shippingAddress) {
+        if (!$shippingAddress = $shippingSubject->getOrder()?->getShippingAddress()) {
```

Comment:
```suggestion
        if (!$shippingSubject instanceof ShipmentInterface::class) {
            return true;
        }

        $shippingAddress = $shippingSubject->getOrder()?->getShippingAddress();

        if ($shippingAddress === null) {
            return true;
        }
```
This way we safely skip non-shipment subjects early without throwing exceptions, keeping the logic simple and BC-friendly.